lib/index.js

```diff
@@ -290,7 +326,7 @@ util.inherits(Device, EventEmitter);
  */
 function createSocket (type, opts) {
     var domain = (opts || {}).raw ? nn.AF_SP_RAW : nn.AF_SP;
-    return new Socket(domain, type);
+    return new Socket(domain, type, opts);
```

Comment:
From simply removing a constructor argument?  You didn't rename the property did you?  see https://github.com/nickdesaulniers/node-nanomsg/blob/master/lib/index.js#L11